subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/taskgrouping/RichVerboseConsoleTypeFunctionalTest.groovy

```diff
@@ -44,22 +48,84 @@ class RichVerboseConsoleTypeFunctionalTest extends AbstractConsoleGroupedTaskFun
                 dependsOn helloWorld, byeWorld, silence
             }
         """
-    }
-
-    @Unroll
-    def "can have verbose task output according to --console"() {
         when:
-        executer.withConsole(mode)
         succeeds('all')
 
         then:
-        result.groupedOutput.task(':helloWorld').output == HELLO_WORLD_MESSAGE
-        result.groupedOutput.task(':byeWorld').output == BYE_WORLD_MESSAGE
+        result.groupedOutput.task(':helloWorld').output == 'Hello world'
+        result.groupedOutput.task(':byeWorld').output == 'Bye world'
         hasSilenceTaskOutput == result.groupedOutput.hasTask(':silence')
 
         where:
         mode    | hasSilenceTaskOutput
         Rich    | false
         Verbose | true
     }
+
+    def "long running task's output are divided by other output"() {
+        given:
+        buildFile << '''
+def sleepFor(String desc, int times){
+    (1..times).each {
+         Thread.sleep(1000)
+         println "${desc} wake up ${it}"
+    }
+}
+
+task('longRunning').doLast {
+    sleepFor('longRunning', 6)
+    assert false
+}
+
+Thread.start {
+    sleepFor('ungrouped', 6)
+}
+'''
+
+        when:
+        fails('longRunning')
+
+        then:
+        result.groupedOutput.task(':longRunning').groupCount == 2
+        result.groupedOutput.task(':longRunning').getStatus(0) == ''
+        result.groupedOutput.task(':longRunning').getStatus(1) == 'FAILED'
```

Comment:
This is a very good assertion. By using a single outcome per group, this assertion won't be explicit. What I would suggest is to assert this in the fixture. In theory and in practice, you can't add more output to a group when the outcome has been set. This means everytime we do `addOutput` in `GroupedOutputFixture`, we would `assert outcome == null`. We can also put a nice message explaining that the outcome of this group has already been identified so no more output should be added for this task.

Reply:
Done.